### ISSUE TITLE
the pre-training loss

### ISSUE THREAD

**Issue by jancylee** [zdaxie/PixPro#9]
The pre-training losses are always negative (like -199.03), is that normal?

**Comment by merlinarer** [comment]
> The pre-training losses are always negative (like -199.03), is that normal?

I got the negative loss also, have you solve it?

**Comment by jancylee** [comment]
Not yet.

**Comment by yinjunbo** [comment]
Have you worked it out? What should be the correct loss?

**Comment by jxhuang0508** [comment]
Anyone reproduced the results or solved the negative loss issue?

**Comment by jianglianEin** [comment]
Anyone konw something about the negative loss? Is that right?

**Comment by impiga** [comment]
Hi, do you mean the value of loss is negative?

If this is the point, this is right for PixPro. You may check the implementation of this line https://github.com/zdaxie/PixPro/blob/main/contrast/models/PixPro.py#L204 .

**Comment by xiao7199** [comment]
Hi, 
If I understand correctly, this regreission loss only includes the pushing force with aligning spatial corresponding pixel. So where is the repulsing force as in Eqn2?

**Comment by ZHEGG** [comment]
why negative loss?

**Comment by shiyongde** [comment]
same question.

**Comment by douyimin** [comment]
I carefully studied the code. The loss of this implementation seems different from the paper. It does not apply comparative loss.

**Comment by douyimin** [comment]
I carefully studied the code. The loss of this implementation seems different from the paper. It does not apply constractive loss.
